anaconda.py

```diff
@@ -691,6 +691,27 @@ def _earlyExceptionHandler(ty, value, traceback):
     from pyanaconda import exception
     anaconda.mehConfig = exception.initExceptionHandling(anaconda)
 
+    # Start the subscription handling thread if the Subscription DBus module
+    # provides enough authentication data.
+    # - as kickstart only supports org + key authentication & nothing
+    #   else currently talks to the Subscription DBus module,
+    #   we only check if organization id & at least one activation
+    #   key are available
+    from pyanaconda.modules.common.util import is_module_available
+    from pyanaconda.modules.common.constants.services import SUBSCRIPTION
+    if is_module_available(SUBSCRIPTION):
+        from pyanaconda.modules.common.structures.subscription import SubscriptionRequest
+        from pyanaconda.modules.common.structures.secret import SECRET_TYPE_HIDDEN
+        from pyanaconda.core.subscription import register_and_subscribe
+        subscription_proxy = SUBSCRIPTION.get_proxy()
+        subscription_request_struct = subscription_proxy.SubscriptionRequest
+        subscription_request = SubscriptionRequest.from_structure(subscription_request_struct)
+        organization_set = bool(subscription_request.organization)
+        key_set = subscription_request.activation_keys.type == SECRET_TYPE_HIDDEN
+        if organization_set and key_set:
+            threadMgr.add(AnacondaThread(name=constants.THREAD_SUBSCRIPTION,
+                                         target=register_and_subscribe))
```

Comment:
Maybe `register_and_subscribe` could have an optional argument that would allow to skip the registration if the organization and the key are not set. Or there could be a new function called `try_register_and_subscribe`. Anyway, it would be good to move this block somewhere else if possible.

Reply:
Indeed, that really should be simpler. So I put that code to function (`org_keys_sufficient()` & `username_password_sufficient()`) that can be imported & used to check if starting the subscription thread makes sense.

I've also used these methods in the `register_and_subscribe()` method to avoid code duplication. In this case they are bit less efficient as they create they own proxies for the Subscription module, but given that this will be called basically once per installation, I don't think that's a problem.